### PR TITLE
chore: make risedev tasks more readable

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- make private
- `--hide-uninteresting`
- group aliases https://github.com/sagiegurari/cargo-make/issues/862
- better categories
- 🌟

```
❯ r+
[cargo-make] INFO - cargo make 0.36.10
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: default
[cargo-make] INFO - Profile: development
Misc
----------
clean-data - Clean all data, config and logs
clean-etcd-data - Clean etcd data
clean-full - Clean all downloaded binaries by deleting .risingwave folder
delete - Delete a node in the cluster [aliases: del]
doc - Open rustdoc for risingwave
download-all - Download all available components at once
follow - Follows the end of the logs of a specific component [aliases: f]
install - 🌟 Install RiseDev so you can run it from anywhere
logs - Open logs with VSCode or dump in console [aliases: l]
ls - List all nodes in the cluster
lsw - List --watch all nodes in the cluster

RiseDev - Build
----------
build-connector-node - Build RisingWave Connector from source
build-docs - Build Rust docs
build-risingwave - Build Rust components [aliases: b, build]
clean - Clean Rust targets
configure - Configure what components to download/build
link-all-in-one-binaries - Link all-in-one cmds to RiseDev bin
sbuild - Build in simulation mode

RiseDev - Check
----------
check - 🌟 Perform pre-CI checks and automatically fix cargo sort, cargo hakari, cargo fmt warnings [aliases: c]
check-clippy - Run cargo clippy check
check-clippy-fix - Run cargo clippy check and fixes all files (including dirty and staged)
check-dep-sort - Run cargo sort check and attempt to fix
check-fix - Perform pre-CI checks and automatically fix cargo sort, cargo hakari, fixes cargo fmt warnings [aliases: cf]
check-fmt - Run cargo fmt check and attempt to fix
check-hakari - Run cargo hakari check and attempt to fix
check-java - Run mvn spotless:check in connector-node
check-java-fix - Run mvn spotless:apply in connector-node
check-typos - Run cargo typos-cli check
check-udeps - Check unused dependencies
install-tools - Install (or upgrade) required tools to do pre-CI check and run e2e tests

RiseDev - Compose
----------
apply-compose-deploy - Run deploy.sh
apply-compose-docker - Run deploy.sh
compose - Compose a docker-compose.yaml file
compose-deploy - Compose a docker-compose.yaml deploy directory

RiseDev - Deterministic Simulation
----------
sarchive-it-test - Archive integration tests in deterministic simulation mode
scheck - Run cargo check in deterministic simulation mode
sit-test - Run integration tests in deterministic simulation mode
sslt - Run e2e tests in deterministic simulation mode
sslt-build-all - Build deterministic simulation runner and tests
sslt-cov - Run e2e tests in deterministic simulation mode and report code coverage
stest - Run unit tests in deterministic simulation mode

RiseDev - Java Binding
----------
build-java-binding-java - Build the java binding java code
build-java-binding-rust - Build the java binding rust code
gen-java-binding-header - Generate the java binding C header file
ingest-data-and-run-java-binding - Ingest test data and run java binding demo
kill-java-binding-demo-cluster - Kill the RisingWave cluster for java binding demo
run-java-binding-demo - Run the java binding demo
run-java-binding-stream-chunk-demo - Run the java binding stream chunk demo
start-java-binding-demo-cluster - Start the RisingWave cluster for java binding demo

RiseDev - SQLLogicTest
----------
docslt - Extract SQL examples written in SQLLogicTest syntax from Rust doc comments
slt - 🌟 Run SQLLogicTest
slt-all - Run all e2e tests
slt-batch - Run all batch e2e tests
slt-generated - Run all generated e2e tests
slt-streaming - Run all streaming e2e tests

RiseDev - Start/Stop
----------
ctl - Start RiseCtl
dev - 🌟 Start a full RisingWave dev cluster using risedev-dev [aliases: d]
down - Kill RisingWave dev cluster and clean the data
kill-risedev - Kill RisingWave dev cluster [aliases: k, kill]
playground - 🌟 Start a lite RisingWave playground using risingwave all-in-one binary [aliases: p]

RiseDev - Test
----------
apply-parser-test - Generate parser test data
compaction-test - Run hummock compaction e2e deterministic test
delete-range-test - Run hummock compaction delete-range for test
do-apply-parser-test - Apply parser test data
do-apply-planner-test - Apply planner test data [aliases: dapt]
do-apply-stream-integration-test - Apply stream integration test output snapshots. [aliases: dasit]
meta-backup-restore-test - Run meta backup/restore test
run-planner-test - Run planner test
test - 🌟 Run unit tests
test-cov - Run unit tests and report coverage

```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
@kwannoel Any idea about fuzz test time out?

image

**Comment by kwannoel** [comment]
> @kwannoel Any idea about fuzz test time out?
> image

I guess there will always be some tail cases which lead to timeout, lets say if every query has natural join. These should occur less often, but they will occur.
Also since we added updates and deletes, the runtime would have increased as well.
I have to look at the set of generated queries for those which take a long time, to know for sure and do some analysis.

For now it is sufficient to just increase the timeout for sqlsmith e2e random fuzz test in `pull-request.yml`, since we don't run it every PR, only PRs with changes to sqlsmith source files / sqlsmith ci scripts.

File: Makefile.toml [hunk 1]
```diff
@@ -997,23 +1018,27 @@ echo If you still feel this is not enough, you may copy $(tput setaf 4)risedev$(
 """
 
 [tasks.ci-start]
+private = true
```
Comment (BugenZhao) [inline Makefile.toml:1021]: From my perspective, I heavily rely on `ci-start` locally for a shortcut of `clean` and `dev`. 🥵

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**